tests/validators/map.test.ts

```diff
@@ -0,0 +1,35 @@
+import { s, ValidationError } from '../../src';
+
+describe('MapValidator', () => {
+	const value = new Map();
+	value.set('a', 1);
+	value.set('b', 2);
+	const predicate = s.map(s.string, s.number);
+
+	test('GIVEN a non-map THEN throws ValidationError', () => {
+		expect(() => predicate.parse(false)).toThrow(new ValidationError('MapValidator', 'Expected a map', false));
+	});
+
+	test('GIVEN a matching map THEN returns a map', () => {
+		expect(predicate.parse(value)).toStrictEqual(value);
+	});
+
+	test('GIVEN a non-matching map THEN throws AggregateError', () => {
+		const map = new Map();
+		map.set('a', 1);
+		map.set('foo', 'bar');
+		map.set(2, 'fizz');
+		map.set(3, 'buzz');
```

Comment:
You can also do
```ts
const map = new Map([
	['a', 1],
	['foo', 'bar'],
	[2, 'fizz'],
	[3, 'buzz']
]);
```

Also you should add a line for:

- 2 valid
- key invalid
- value invalid
- 2 invalid

That way we can check that it's adding both keys

Reply:
Setting it in the constructor throws a TS error because of the mismatched types, so I can either `@ts-ignore` it, or stick with `set`.